Compilation failure: KeycloakModelSchema cannot find symbol KeycloakModelSchemaImpl

### Before reporting an issue

- [x] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

infinispan

### Describe the bug

Building `model/infinispan` on OpenJDK 24 EA results in a compilation failure:

```
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /home/st/dev/keycloak/model/infinispan/src/main/java/org/keycloak/marshalling/KeycloakModelSchema.java:[237,40] cannot find symbol
  symbol:   class KeycloakModelSchemaImpl
  location: interface org.keycloak.marshalling.KeycloakModelSchema
```

### Version

main

### Regression

- [ ] The issue is a regression

### Expected behavior

It should be possible to build Keycloak with OpenJDK 24

### Actual behavior

Not possible to build Keycloak with OpenJDK 24

### How to Reproduce?

Run `mvn clean install -DskipTests` with OpenJDK 24.

### Anything else?

_No response_

--- a/model/infinispan/pom.xml
+++ b/model/infinispan/pom.xml
@@ -119,6 +119,24 @@
                     <remoteLockFiles>https://raw.githubusercontent.com/keycloak/keycloak/refs/heads/release/26.0/model/infinispan/proto.lock,https://raw.githubusercontent.com/keycloak/keycloak/refs/heads/release/26.1/model/infinispan/proto.lock</remoteLockFiles>
                 </configuration>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <annotationProcessorPaths>
+                        <path>
+                            <groupId>org.infinispan</groupId>
+                            <artifactId>infinispan-component-processor</artifactId>
+                            <version>${infinispan.version}</version>
+                        </path>
+                        <path>
+                            <groupId>org.infinispan.protostream</groupId>
+                            <artifactId>protostream-processor</artifactId>
+                            <version>${protostream.version}</version>
+                        </path>
+                    </annotationProcessorPaths>
+                </configuration>
+            </plugin>
         </plugins>
     </build>
 </project>

--- a/pom.xml
+++ b/pom.xml
@@ -54,6 +54,7 @@
 
         <quarkus.version>3.19.0.CR1</quarkus.version>
         <quarkus.build.version>3.19.0.CR1</quarkus.build.version>
+        <jboss.logging.version>3.0.3.Final</jboss.logging.version> <!-- keep in sync with the version used by quarkus -->
 
         <project.build-time>${timestamp}</project.build-time>
 
@@ -97,7 +98,8 @@
         <hibernate-orm.plugin.version>6.2.13.Final</hibernate-orm.plugin.version>
         <hibernate.c3p0.version>6.2.13.Final</hibernate.c3p0.version>
         <infinispan.version>15.0.14.Final</infinispan.version>
-        <protostream.plugin.version>5.0.13.Final</protostream.plugin.version>
+        <protostream.version>5.0.13.Final</protostream.version> <!-- For the annotation processor: keep in sync with the version shipped with Infinispan -->
+        <protostream.plugin.version>${protostream.version}</protostream.plugin.version>
 
         <!--JAKARTA-->
         <jakarta.mail.version>2.1.1</jakarta.mail.version>

--- a/services/pom.xml
+++ b/services/pom.xml
@@ -265,6 +265,13 @@
                     <compilerArgument>
                         -AgeneratedTranslationFilesPath=${project.build.directory}/generated-translation-files
                     </compilerArgument>
+                    <annotationProcessorPaths>
+                        <path>
+                            <groupId>org.jboss.logging</groupId>
+                            <artifactId>jboss-logging-processor</artifactId>
+                            <version>${jboss.logging.version}</version>
+                        </path>
+                    </annotationProcessorPaths>
                 </configuration>
             </plugin>
             <plugin>

--- a/test-framework/bom/pom.xml
+++ b/test-framework/bom/pom.xml
@@ -83,12 +83,6 @@
             </dependency>
             <dependency>
                 <groupId>org.keycloak.testframework</groupId>
-                <artifactId>keycloak-test-framework-oauth</artifactId>
-                <version>${project.version}</version>
-                <scope>test</scope>
-            </dependency>
-            <dependency>
-                <groupId>org.keycloak.testframework</groupId>
                 <artifactId>keycloak-test-framework-email-server</artifactId>
                 <version>${project.version}</version>
                 <scope>test</scope>

